src/utils/types.js

```diff
@@ -0,0 +1,10 @@
+const type = {
+  isArray: (value) => value && typeof value === 'object' && value.constructor === Array,
+  isBoolean: (value) => typeof value === 'boolean',
+  isFunction: (value) => typeof value === 'function',
+  isNumber: (value) => typeof value === 'number' && Number.isFinite(value),
+  isObject: (value) => typeof value === 'object' || value.constructor === Object,
+  isString: (value) => typeof value === 'string' || value instanceof String,
```

Comment:
The extra check is redundant for the same reason as above. Any value that is an instance of `String` will have `typeof value === 'string'`. I can probably think of a few ways to produce a value that is an instance of `String` with a non-string `typeof`, but again, I wouldn't call those values strings anyway.

Reply:
It's to handle strings created by `new String()`:

```js
const checks1 = [new String('test')].map((value) => {
	return typeof value === 'string' || value instanceof String;
});


const checks2 = [new String('test')].map((value) => {
	return typeof value === 'string';
});

console.log(checks1);
console.log(checks2);
```

Output:

```
Array [ true ]
Array [ false ]
```